### PR TITLE
Adds Clamp Function

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

Adds a clamp function to the default functions. This is essentially a shortcut for doing:
```
set {_value} to min(max({_value}, {_min}), {_max})
```
It returns the given value if it's between the min/max.
It returns the max if the given value is above the max.
It returns the min if the given value is below the min.
```
set {_value} to clamp({_value}, {_min}, {_max})
```
Also, no clue how the indentation is supposed to go, since the whole thing is just one long run-on line, so I just followed what all the other functions did.

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:**

### CODE REVIEW

File: src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java [hunk 1]
```diff
@@ -305,7 +304,37 @@ public Number[] executeSimple(Object[][] params) {
...
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("clamp", new Parameter[]{
+			new Parameter<>("values", DefaultClasses.NUMBER, false, null),
+			new Parameter<>("min", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("max", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, false) {
+			@Override
+			public @Nullable Number[] executeSimple(Object[][] params) {
+				Number[] values = (Number[]) params[0];
+				Double[] clamped_values = new Double[values.length];
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java:316]: Change the name of `clampled_values` to `clampedValues`

File: src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java [hunk 2]
```diff
@@ -305,7 +304,37 @@ public Number[] executeSimple(Object[][] params) {
...
+				double min = ((Number) params[1][0]).doubleValue();
+				double max = ((Number) params[2][0]).doubleValue();
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java:318]: Maybe check if these values are null

**Review by UnderscoreTud — CHANGES_REQUESTED** [review]

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java [hunk 3]
```diff
@@ -305,7 +304,37 @@ public Number[] executeSimple(Object[][] params) {
...
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("clamp", new Parameter[]{
+			new Parameter<>("values", DefaultClasses.NUMBER, false, null),
+			new Parameter<>("min", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("max", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, false) {
+			@Override
+			public @Nullable Number[] executeSimple(Object[][] params) {
+				Number[] values = (Number[]) params[0];
+				Double[] clampedValues = new Double[values.length];
```
Comment (UnderscoreTud) [inline src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java:316]: Could this be the primitive type? `double[]`

**Review by UnderscoreTud — APPROVED** [review]

File: src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java (hunk 3 above)
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java:316]: I don't think so, it wants a Number[] as the return value so Double[] is fine but double[] is not.

Comment (UnderscoreTud) [reply src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java:316]: Ohh, you're right. it's fine how it is now then

**Pull request merged**